### PR TITLE
[JIT][NNC] add hardswish symbolic gradient and NNC lowering

### LINKED ISSUE #56969: [JIT / NNC] Incomplete Coverage of Activations
## 🐛 Bug

We have run into internal and external models which are missing fusion opportunities because we a) don't support all activations in autodiff and b) don't support all activations in NNC. These are very easy to add and important for generating fusions. 

[List of Pointwise Activations](https://pytorch.org/docs/stable/nn.html#non-linear-activations-weighted-sum-nonlinearity):

Symbolic Script Support (necessary for autodiff):

- [ ] ELU
- [ ] Hardshrink
- [ ] Hardsigmoid
- [ ] Hardtanh
- [ ] Hardswish
- [ ] LeakyRelu
- [ ] LogSigmoid
- [ ] PReLU
- [x] ReLU
- [ ] ReLU6
- [ ] RReLU
- [ ] SELU
- [ ] CELU
- [ ] GELU
- [x] Sigmoid
- [ ] SiLU
- [ ] Softplus
- [ ] Softshrink
- [ ] Softsign
- [x] Tanh
- [ ] Tanhshrink
- [x] Threshold

NNC Support (necessary for fusion)

- [ ] ELU
- [ ] Hardshrink
- [ ] Hardsigmoid
- [x] Hardtanh
- [ ] Hardswish
- [ ] LeakyRelu
- [ ] LogSigmoid
- [ ] PReLU
- [x] ReLU
- [ ] ReLU6
- [ ] RReLU
- [ ] SELU
- [ ] CELU
- [ ] GELU
- [x] Sigmoid
- [ ] SiLU
- [ ] Softplus
- [ ] Softshrink
- [ ] Softsign
- [x ] Tanh
- [ ] Tanhshrink
- [x] Threshold

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#57383 [JIT][NNC] add hardswish symbolic gradient and NNC lowering**

Notes: I picked up an activation from https://github.com/pytorch/pytorch/issues/56969. You can look at the [activations.cpp](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/Activation.cpp#L429) file which has both forward and backward kernel code to help you write the NNC lowering and the symbolic gradient. 

I added a test in test_jit_fuser_te for the fusion, and I added an OpInfo and asserted that we expect to see autodiffable nodes to test the symbolic gradient.

Differential Revision: [D28197820](https://our.internmc.facebook.com/intern/diff/D28197820)

### CODE REVIEW

File: torch/csrc/jit/runtime/symbolic_script.cpp [hunk 1]
```diff
@@ -846,6 +846,15 @@ const std::vector<std::string> functions = {
 
             return result, backward
 
+        def hardswish(self):
+            result = torch.relu(self)
+            def backward(grad_output):
+                m = (result >= 3.).type_as(result)
+                m = torch.where((result >= -3.) & (result <= 3.),  result / 3. + .5, m)
```
Comment (penguinwu) [inline torch/csrc/jit/runtime/symbolic_script.cpp:853]: This is neat using `relu(self)>3.` to compute the derivative of hardswish when self is out of the range of (-3,3).

But shouldnt the condition be `(result > -3.) & (result < 3.)`?

**Review by penguinwu — APPROVED** [review]

File: torch/csrc/jit/runtime/symbolic_script.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/runtime/symbolic_script.cpp:853]: ooh i think i must have bad copy-pasta'd in the definition of hardswish (thus the test failures). should be `result = torch.hardswish(relu)`. 

Looks like it should be one >= and one < judging by the activations file https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/Activation.cpp#L466

Comment (eellison) [reply torch/csrc/jit/runtime/symbolic_script.cpp:853]: Okay so actually, it doesn't matter at x == 3, since if `x = 3` then `x * (x + 3)/6 == x`. It does matter when x = -3, and that's why we have the result >= -3. condition. However it would be clearer if I rewrote `m = (result >= 3.).type_as(result)` to be `m = (result >= 3.).type_as(result)`.

Comment (penguinwu) [reply torch/csrc/jit/runtime/symbolic_script.cpp:853]: Interesting, if we use the formula of `hardswish(x) = x` when x=3, the derivative is 1; but if we use `hardwish(x)= x(x+3)/6`, the derivative is `x/3+.5` (which is 1.5). I guess derivative at the boundary point is not precise.

**Comment by mrshenli** [comment]
This commit breaks master. Reverting.
